diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 8e02ea6ca2..3b4ca12072 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -266,6 +266,13 @@
 ; Remote Settings JSON dumps
 @RESPATH@/browser/defaults/settings
 
+; FireDragon settings
+#ifdef FIREDRAGON_SETTINGS
+@RESPATH@/defaults/pref/local-settings.js
+@RESPATH@/distribution/policies.json
+@RESPATH@/firedragon.cfg
+#endif
+
 # channel-prefs.js has been removed on macOS.
 #ifndef XP_MACOSX
 ; Warning: changing the path to channel-prefs.js can cause bugs (Bug 756325)
