diff --git a/browser/moz.configure b/browser/moz.configure
index c4184aa4fa..36b2506efd 100644
--- a/browser/moz.configure
+++ b/browser/moz.configure
@@ -26,7 +26,7 @@ imply_option("MOZ_DEVTOOLS", "all")
 imply_option("BROWSER_CHROME_URL", "chrome://browser/content/browser.xhtml")
 
 ## Noraneko Patch
-include("../floorp/gecko/config/moz.configure")
+include("../firedragon/gecko/config/moz.configure")
 ## End of Noraneko Patch
 
 option(
