diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index d069529341..4648a80dca 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -193,6 +193,15 @@
 @RESPATH@/chrome/remote.manifest
 #endif
 
+; [Noraneko Files]
+#ifdef NORANEKO_DIST
+@RESPATH@/chrome/noraneko.manifest
+@RESPATH@/chrome/noraneko@JAREXT@
+#endif
+#ifdef NORANEKO_BUILDID2
+@RESPATH@/browser/buildid2
+#endif
+
 ; [Extensions]
 @RESPATH@/components/extensions-toolkit.manifest
 @RESPATH@/browser/components/extensions-browser.manifest
@@ -263,6 +272,13 @@
 ; Remote Settings JSON dumps
 @RESPATH@/browser/defaults/settings
 
+; [FireDragon Settings]
+#ifdef FIREDRAGON_SETTINGS
+@RESPATH@/defaults/pref/local-settings.js
+@RESPATH@/distribution/policies.json
+@RESPATH@/firedragon.cfg
+#endif
+
 # channel-prefs.js has been removed on macOS.
 #ifndef XP_MACOSX
 ; Warning: changing the path to channel-prefs.js can cause bugs (Bug 756325)
@@ -398,10 +414,6 @@ bin/libfreebl_64int_3.so
 @BINPATH@/crashhelper@BIN_SUFFIX@
 #endif
 
-; [ Ping Sender ]
-;
-@BINPATH@/pingsender@BIN_SUFFIX@
-
 ; [ Native Messaging Host Proxy ]
 ;
 #if defined(XP_WIN) || defined(XP_MACOSX)
